### DISCUSSION TITLE
Kafka Users are created with ACL entries and during performing operations allowed by ACL we see Denied Operation

### DISCUSSION THREAD

**Discussion by Janardhan78** [strimzi/strimzi-kafka-operator#10102] [General]
We are using 0.36.1 version of strimzi and 3.5.1 version of kafka 

We have created 2 KafkaUsers from Strimzi operator for 2 different cluster with ACL entries. we have observed that the ACL entries in teh Kafka Cluster will not be present for approximately 4 minutes  and the ACL entries will be available after that. The clients which are using KafkaUser to perform operations not be able to perform operations till the ACL is available in KafkaCLuster.

We see in the Kafka Logs below are the mesages the user and cluster details in messages are not added since it is proprietery

Processing notification(s) to /config/change

Processing override for entityPath

Removing PRODUCE quota for user

Removing FETCH quota for user

Removing REQUEST quota for user

Removing CONTROLLER_MUTATION quota for user

Processing notification(s) to /kafka-acl-changes

Processing Acl change notification for

Processing notification(s) to /config/changes

Processing notification(s) to /kafka-acl-changes

Processing Acl change notification for ResourcePattern(resourceType=GROU

 

The same behavior is observed even if we create a single KafkaUser , same behavior is observed

 

 We tried to check by creating a ACL in Kafka Cluster we see that ACL is getting removed in kafkacluster. 

Even if we try to add a ACL manually from kafka cluster the same behavior is seen. After some time the ACL is getting deleted. Have raised a Bug in ASF https://issues.apache.org/jira/browse/KAFKA-16693. I havent received any response for the same. Has any one facing similar issue ?

 

kafka-acls --add --allow-host '*' --allow-principal User:'CN=jana-2' --operation Describe --resource-pattern-type LITERAL --topic my-new-topic --bootstrap-server eric-data-message-bus-kf-9-kafka-bootstrap:9091 -command-config /tmp/kafka_ssl_config.properties 
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Adding ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

 

ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:24:53 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:24:58 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:02 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:06 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:10 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:15 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:19 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:24 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:28 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:32 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:36 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:40 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:45 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:49 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:53 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:25:58 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:02 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:06 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:10 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:15 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:19 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:23 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:28 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:32 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-new-topic, patternType=LITERAL)`: 
     (principal=User:CN=jana-2, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Mon May 13 14:26:36 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Mon May 13 14:26:40 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Mon May 13 14:26:45 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
^CMon May 13 14:26:48 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Mon May 13 14:26:52 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'

**Comment by scholzj** [comment]
Please share the full logs and custom resources and properly explain the steps you are doing as I do not think they are really clear.

**Reply by Janardhan78** [reply]
Hi @scholzj 
I am not able to share the logs since it is proprietary. 
We could reproduce it by 
1. create a kafka cluster using strimzi operator
2. create a kafkauser with ACL defined
3. after creating the kafka user login to one of the kafka broker 
4. list the acl in the kafka cluster, The created ACL will be present for some time and after some time it will be deleted. During the duration of the ACL being deleted, any client cannot perform the Operations allowed in ACL
    
    bash-4.4$ while true 
> do 
> kafka-acls --bootstrap-server eric-data-message-bus-kf-9-kafka-bootstrap:9091 -command-config /tmp/kafka_ssl_config.properties --list
> date
> done
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:05:53 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:05:58 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:02 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:06 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:11 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:15 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:20 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:24 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:29 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:33 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Current ACLs for resource `ResourcePattern(resourceType=TOPIC, name=my-topic, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=WRITE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=CREATE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=DESCRIBE, permissionType=ALLOW)
	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Current ACLs for resource `ResourcePattern(resourceType=GROUP, name=my-group, patternType=LITERAL)`: 
 	(principal=User:CN=my-mbkf-user, host=*, operation=READ, permissionType=ALLOW) 

Tue May 14 15:06:38 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:06:42 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:06:47 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:06:51 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:06:56 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:07:00 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:07:05 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:07:09 UTC 2024
ERROR StatusLogger Reconfiguration failed: No configuration found for '2cdf8d8a' at 'null' in 'null'
Tue May 14 15:07:13 UTC 202

**Reply by scholzj** [reply]
> I am not able to share the logs since it is proprietary.

Well, that we might not be able to help you. 

> We could reproduce it by

You have to provide logs or a proper reproducer. The reproducer should include exact steps for reproducing it including the exact custom resources and configs you are using. You should also not run any commands from the Kafka broker pods -> you should setup your own pods and use proper authentication for them. The port 9091 is only for the internal communication of the Kafka cluster and you likely hacked around the configuration to connect to it.

**Reply by Janardhan78** [reply]
Hi @scholzj 

I tried to check if i can provide reproducer for the scenario i have explained. but i am not able to provide since there is few customizations that is done in the strimzi operator more specific to suit our customer needs. 

1. kafka cluster of version 3.5.1 using strimzi operator 0.36.1
2. create a kafkauser for the cluster with ACL entries
3. The ACL entries will be deleted and after some times it will be created during reconcilation from user operator.

is it possible to check this 

Regards
Jana

**Reply by scholzj** [reply]
But this works completely fine -> it creates the ACLs as configured in the KafkaUser resource and keeps them configured. This is also part of our system tests executed for every release.

To create a reproducer, you do not have to share anything private or custom. That would be actually detrimental for a reproducer. Just start a new fresh Minikube or Kind cluster and install latest Strimzi version (because nobody is going to fix any issues in Strimzi 0.36 anymore). And then use the Kafka and KafkaUser resources with only the configuration needed to reproduce the problem and provide the resources and the exact steps you used. This is actually important for the reproducer, because it helps you check that the issue is in Strimzi and not in your secret configuration and the private customizations you are using.

**Reply by Janardhan78** [reply]
Hi @scholzj 

I tried to setup strimzi 0.36.1 in my environmet using minikube , was not able to reproduce the error condition. The ACL are present in the kafka cluster. Will check in our custom deployment to see the reason for deletion

<img width="629" alt="acl_list" src="https://github.com/strimzi/strimzi-kafka-operator/assets/69192936/254a0ade-4689-434b-b1c0-3ff4bfd26342">
<img width="483" alt="kafak-2-logs" src="https://github.com/strimzi/strimzi-kafka-operator/assets/69192936/36cc7892-abf8-4167-a57b-683a8a249e9d">
<img width="881" alt="kafka_1_logs" src="https://github.com/strimzi/strimzi-kafka-operator/assets/69192936/6adf2119-9a40-48af-978b-1cf9c0105786">
<img width="483" alt="kafka-0-logs" src="https://github.com/strimzi/strimzi-kafka-operator/assets/69192936/456d34d0-2441-47db-b969-51b27b98dbc2">
[kakfa_test_logs.zip](https://github.com/strimzi/strimzi-kafka-operator/files/15354729/kakfa_test_logs.zip)
